sympy/functions/elementary/piecewise.py

```diff
@@ -237,6 +238,32 @@ def _eval_derivative(self, x):
     def _eval_evalf(self, prec):
         return self.func(*[(e._evalf(prec), c) for e, c in self.args])
 
+    def _eval_is_meromorphic(self, x, a):
+        # Conditions often implicitly assume that the argument is real.
+        # Hence, there needs to be some check for as_set.
+        if not a.is_real:
+            return None
+
+        a_real = a.is_real
+        # First, let's check if a is real. If not return ValueError.
+        if a_real is False:
+            raise ValueError("A piecewise function is defined only on the real line.")
+        if a_real is None:
+            return None
```

Comment:
These lines are not needed now that `a.is_real` is tested above. `a_real` will be `True` and these checks will never succeed. So this block should be removed.